src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DeleteCompleteFilesTask.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks;
+
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.BlobManager;
+import uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor.EnvelopeProcessor;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.bulkscanprocessor.entity.Status.COMPLETED;
+
+@Service
+public class DeleteCompleteFilesTask {
+    private static final Logger log = LoggerFactory.getLogger(DeleteCompleteFilesTask.class);
+
+    private static final String TASK_NAME = "delete-complete-files";
+
+    private final BlobManager blobManager;
+    private final EnvelopeRepository envelopeRepository;
+    private final EnvelopeProcessor envelopeProcessor;
```

Comment:
is `EnvelopeProcessor` needed
or would `EnvelopeRepository` be sufficient?

Reply:
This is from existing code, EnvelopeProcessor is in charge of handling DataIntegrityViolationException

Reply:
I think it's there for adding new envelopes

Reply:
No, it is called from BlobProcessorTask::deleteIfCompleted

Reply:
I meant that the `DataIntegrityViolationException` handling is there for saving new envelopes